apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -1538,10 +1538,22 @@ export async function render(
         scriptUrls.push(assets.coreElementAssetPath(file)),
       );
       dependencies.courseElementStyles.forEach((file) =>
-        styleUrls.push(assets.courseElementAssetPath(course.commit_hash, locals.urlPrefix, file)),
+        styleUrls.push(
+          assets.courseElementAssetPath(
+            course.commit_hash,
+            locals.urlPrefix + question.course_id !== course.id ? `/variant/${variant.id}` : '',
```

Comment:
I believe you want brackets here (addition has higher precedence than ternary conditional):
```suggestion
            locals.urlPrefix + (question.course_id !== course.id ? `/variant/${variant.id}` : ''),
```

Reply:
That said: is there any advantage with using path parameters vs query parameters here? In other words, why not use `/pl/course/XXX/cacheableElements/YYY?variant_id=ZZZ`? That would reduce the need for changes in server.js. Unless you want to do some authorization check based on variants and prefixes, which is currently not done.

Reply:
Another issue: this is done for any question, not just shared questions, which means the caching of elements in local browsers becomes linked to the variant, and you lose a lot of potential for caching.

Reply:
ah shoot, I totally missed the cacheing thing. Ya we should probably do this differently.

My main concern, aren't query params usually for smaller changes to what an endpoint means? Having the files come from a different course seems like a big change that should be a path param. But I guess I don't really know what our conventions are around path vs. query params.

Reply:
All fixed.